libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance.hs

```diff
@@ -418,3 +504,140 @@ instance
+      , " ∀[ (a , _) ∈ range txouts ] Sum.All (const ⊤) (λ a → a .BootstrapAddr.attrsSize ≤ 64) a\n  "
+          -- I assume we never create bootstrap addresses
+          <> show (all (\txout -> case txout ^. addrTxOutL of AddrBootstrap _ -> False; _ -> True) txouts)
+      ]
+  logEntry $
+    unlines
+      [ "Implementation:"
+      , "  consumed:\t" ++ show (coin consumedVal)
+      , "    ins:\t\t" ++ show (coinBalance (UTxO (Map.restrictKeys (unUTxO utxo) (txb ^. inputsTxBodyL))))
+      , "  produced:\t" ++ show (coin producedVal)
+      , "    outs:\t" ++ show (coin $ sumAllValue txouts)
+      , "    fee:\t\t" ++ show txfee
+      ]
+  lastTick <- gets $ view impLastTickL
+  pParams <- getsNES $ nesEsL . curPParamsEpochStateL
+  certState <- getsNES $ nesEsL . esLStateL . lsCertStateL
+  let
+    utxoEnv =
+      UtxoEnv
+        { ueSlot = lastTick
+        , uePParams = pParams
+        , ueGenDelegs = certState ^. certDStateL . dsGenDelegsL
+        , ueCertState = certState
+        }
+  agdaUtxoEnv <- case toSpecRep utxoEnv of
+    Right x -> pure x
+    Left e -> fail $ T.unpack e
+  let
+    newAgdaUtxoState = Agda.utxoStep agdaUtxoEnv agdaUtxoState agdaTx
+  modify $ impDoFixupL .~ False
+  submitRes <- trySubmitTx tx
+  modify $ impDoFixupL .~ doFixup
+  newUtxoState <- getsNES $ nesEsL . esLStateL . lsUTxOStateL
+  let
+    finalAgdaState =
+      toExpr . specToTestRep @(UTxOState Conway) <$> newAgdaUtxoState
+    finalImplState = case submitRes of
+      Right _ -> case toTestRep newUtxoState of
+        Right x -> Just $ toExpr x
+        Left _ -> Nothing
+      Left e ->
+        error $
+          "Failed to submit transaction on the implementation side:\n"
+            <> show e
+  unless (finalAgdaState == finalImplState) $ do
+    logEntry $ diffExprCompact (toExpr finalAgdaState) (toExpr finalImplState)
+    fail "The final states of spec and implementation do not match"
+  pure submitRes
+
+spec :: HasCallStack => Spec
+spec = describe "Conway conformance tests" . withImpState $ do
+  it "Simple tx" $ do
+    modify $ \x ->
+      x
+        & impLastTickL .~ 100
+    do
+      certState <- getsNES $ nesEsL . esLStateL . lsCertStateL
+      govState <- getsNES $ nesEsL . esLStateL . lsUTxOStateL . utxosGovStateL
+      totalObligation certState govState `shouldBe` zero
+    do
+      deposited <- getsNES $ nesEsL . esLStateL . lsUTxOStateL . utxosDepositedL
+      deposited `shouldBe` zero
+    res <- trySubmitTxConform $ mkBasicTx mkBasicTxBody
+    case res of
+      Right _ -> pure ()
+      Left e -> error $ show e
+  xit "Constrained generator" $ do
+    let
+      us =
+        def
+          { usMinCerts = 0
+          , usMaxCerts = 0
+          , usDatumFreq = 0
+          , usGenerateWithdrawals = False
+          }
+    modify $ \x ->
+      x
+        & impLastTickL .~ 100
+        & impDoFixupL .~ False
+    (nes, tx, _env) <- liftIO . generate . genTxAndNewEpoch us $ reify @Conway
```

Comment:
Don't use `generate` outside debugging things in `ghci` or something
```suggestion
```

I know this is not the end of work on conformance tests, but still, I think we need to solve data generation first as part of this PR